src/components/media-views.js

```diff
@@ -890,6 +899,12 @@ AFRAME.registerComponent("media-image", {
 
     const projection = this.data.projection;
 
+    if (this.mesh && this.data.batch) {
+      // This is a no-op if the mesh was just created.
+      // Otherwise we want to ensure the texture gets updated.
+      batchManagerSystem.removeObject(this.mesh);
```

Comment:
Can't you just do this above where `textureCache.release(src, version);` is being called.

Reply:
I don't think so, that release i think is handling the specific case where one of the promises is still resolving since the top of the same function call, where the local `src` and `version` were bound. in this case, we want to un-bind any existing texture even it was bound in a previous call site.

the other place to do it was at the very beginning when we check for the src/version to have changed, which I tried, but the problem there is that it results in the texture blinking for a second, as this function yields to several promises between here and there. by doing the remove later (here), the swap happens (ideally) intra-frame.